manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -52,6 +52,11 @@ spec:
           operator: Exists
         - effect: NoExecute
           operator: Exists
+      securityContext:
+        runAsGroup: 0
+        runAsUser: 0
+        runAsNonRoot: false
+        readOnlyRootFilesystem: false
```

Comment:
Do we need this actually? this doesn't block us from writing to the hostPath mounts

Reply:
readOnlyRootFilesystem is by default `false`. Also, do we need to set the securityContext at the pod level or is it enough to set it at the container level - specifically just for the install-cni container? Can the taint-controller container run as a non-root user? In my local dev env, I can run this with the securityContext just set for the install-cni container.

`mariamj@tryma1:~$ kubectl get pods -n istio-system
NAME                                   READY   STATUS    RESTARTS   AGE
istio-cni-node-vcx94                   2/2     Running   0          29s
istio-ingressgateway-d988d997f-mlftn   1/1     Running   0          29s
istiod-784cd69459-gppk9                1/1     Running   0          35s`

`kubectl exec -it istio-cni-node-vcx94 -n istio-system --container taint-controller -- /bin/sh
$ whoami
istio-proxy
$ exit
mariamj@tryma1:~$ kubectl exec -it istio-cni-node-vcx94 -n istio-system --container install-cni -- /bin/sh
# whoami
root`